### PR TITLE
Fix error when hovering over non-emoji reactions

### PR DESCRIPTION
**Pull request by robintown**
Closes https://github.com/vector-im/element-web/issues/18107.

element-web notes: none

### CODE REVIEW

File: src/HtmlUtils.tsx [hunk 1]
```diff
@@ -79,8 +79,8 @@ function mightContainEmoji(str: string): boolean {
  * @return {String} The shortcode (such as :thumbup:)
  */
 export function unicodeToShortcode(char: string): string {
-    const shortcodes = getEmojiFromUnicode(char).shortcodes;
-    return shortcodes.length > 0 ? `:${shortcodes[0]}:` : '';
+    const shortcodes = getEmojiFromUnicode(char)?.shortcodes;
+    return shortcodes && shortcodes.length > 0 ? `:${shortcodes[0]}:` : '';
```
Comment (SimonBrandner) [inline src/HtmlUtils.tsx:83]:
```suggestion
    return shortcodes?.length > 0 ? `:${shortcodes[0]}:` : '';
```
Shouldn't this work?

Comment (robintown) [reply src/HtmlUtils.tsx:83]: I think so, but it's helpful for readability to not have to contemplate whether `undefined > 0` :dizzy:

Comment (t3chguy) [reply src/HtmlUtils.tsx:83]: I tend to do just do foo?.length and skip the comparison as 0 is falsey and 1+ is truthy

**Review by t3chguy — APPROVED** [review]
Great, thanks!

**Pull request merged**